### PR TITLE
Enable also repositories without `.git`sufix to be seen

### PR DESCRIPTION
**Pull request by mltbnz**
We use your tool and it is a nice but was ignoring a couple of our dependencies.

Screen Shot 2021-11-17 at 19 17 30

I figured that all repos without the `.git` suffix were ignored but packages with a url e.g. `https://github.com/kylef/Commander` can also be added to Xcode as a valid package.

Because of that I removed that check from the guard here 748a1cb05806e1a12c552b2f5b23ae3408735652 which works great for us. But I fear it could create other issues further down the road. Maybe you can clarify

Besides that I broke down the package a bit to first and foremost enable tests since that was not supported with the `.executable` product. And of course added a test for the component behavior change.

### CODE REVIEW

**Comment by luizmb** [comment]
Hey @mltbnz,

Great work! I'm gonna review your changes and test against all of our projects asap.

File: Sources/Models/GitHub.swift [hunk 1]
```diff
@@ -39,18 +40,21 @@ func githubRepository(from url: URL) -> Result<GitHubRepository, GeneratePlistEr
 
     guard let host = url.host,
         host.contains(gitDomain),
-        url.pathComponents.count >= 2,
-        url.pathComponents[url.pathComponents.count - 1].hasSuffix(gitSuffix) else { return .failure(.unknownRepository) }
+        url.pathComponents.count >= 2
+    else { return .failure(.unknownRepository) }
 
+    let name = url.pathComponents.last!.contains(gitSuffix)
```
Comment (luizmb) [inline Sources/Models/GitHub.swift:46]: Even if the .count >= 2, please don't force unwrap, instead make `let lastPathComponent = url.pathComponents.last` in the `guard` condition.

File: Sources/Models/GitHub.swift [hunk 2]
```diff
@@ -39,18 +40,21 @@ func githubRepository(from url: URL) -> Result<GitHubRepository, GeneratePlistEr
...
+        ? String(url.pathComponents[url.pathComponents.count - 1].dropLast(gitSuffix.count))
+        : String(url.pathComponents[url.pathComponents.count - 1].replacingOccurrences(of: gitSuffix, with: ""))
```
Comment (luizmb) [inline Sources/Models/GitHub.swift:48]: In the motto of ,better safe than sorry', we could use `[safe: ]` subscript every time we use array subscript in this file.

**Review by luizmb — APPROVED** [review]
Some minor comments, but generally speaking it looks good to me!
I'm gonna test now with our projects.

File: Sources/Models/GitHub.swift (hunk 2 above)
Comment (mltbnz) [reply Sources/Models/GitHub.swift]: I think we can also use the `lastPathComponent` then which reads even nicer imo.
for the subscript: I added one to Helpers now (or is there something I can use already. Swift itself doesn't offer that, right?) 
Added an additional guard but not sure if throwing the correct error?

File: Sources/Helper/Array+Safe.swift [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+extension Array {
+    public subscript(index: Int, default defaultValue: @autoclosure () -> Element) -> Element {
```
Comment (luizmb) [inline Sources/Helper/Array+Safe.swift:4]: Oops, I forgot to tell you, this extension is already defined in here (I believe Helper already depends on FoundationExtensions):

https://github.com/teufelaudio/FoundationExtensions/blob/master/Sources/FoundationExtensions/Collection/Collection%2BExtensions.swift#L77

There's no `defaultValue` parameter, but a simple `??` operator would solve that.

**Review by luizmb — APPROVED** [review]

File: Sources/Helper/Array+Safe.swift (hunk 3 above)
Comment (mltbnz) [reply Sources/Helper/Array+Safe.swift]: alright. thanks. Some handy tooling

**Pull request merged**